datalad_next/tests/test_create_sibling_webdav.py

```diff
@@ -368,3 +371,136 @@ def query(self, *args, **kwargs):
+                 'f3': '3'})
+@with_tempfile
+@serve_path_via_webdav(auth=webdav_cred[1:])
+def test_existing_switch(localpath, remotepath, url):
+    ca = dict(result_renderer='disabled')
+    ds = Dataset(localpath).create(force=True, **ca)
+    sub = ds.create('sub', force=True, **ca)
+    sub2 = ds.create('sub2', force=True, **ca)
+    subsub = sub2.create('subsub', force=True, **ca)
+    ds.save(recursive=True)
+
+    # need to amend the test credential, can only do after we know the URL
+    ds.credentials(
+        'set',
+        name='dltest-mywebdav',
+        # the test webdav webserver uses a realm label '/'
+        spec=dict(realm=url + '/'),
+        **ca)
+
+    subsub.create_sibling_webdav(f'{url}/sub2/subsub', storage_sibling='yes',
+                                 **ca)
+    sub2.create_sibling_webdav(f'{url}/sub2', storage_sibling='only', **ca)
+    sub.create_sibling_webdav(f'{url}/sub', storage_sibling='no', **ca)
+
+    res = ds.create_sibling_webdav(f'{url}', storage_sibling='yes',
+                                   existing='skip',
+                                   recursive=True, **ca)
+    dlaurl='datalad-annex::?type=webdav&encryption=none&exporttree=no&' \
+           'url=http%3A//127.0.0.1%3A43612/'
+
+    # results per dataset:
+    # super:
+    assert_in_results(
+        res,
+        action='create_sibling_webdav.storage',
+        status='ok',
+        type='dataset',
+        path=ds.path
+    )
+    assert_in_results(
+        res,
+        action='add-sibling',
+        status='ok',
+        type='sibling',
+        path=ds.path,
+        name='127.0.0.1',
+        url=dlaurl[:-1],
+    )
+    # sub
+    assert_in_results(
+        res,
+        action='create_sibling_webdav.storage',
+        status='ok',
+        type='dataset',
+        path=sub.path
+    )
+    assert_in_results(
+        res,
+        action='create_sibling_webdav',
+        status='notneeded',
+        type='dataset',
+        path=sub.path,
+    )
+    # sub2
+    assert_in_results(
+        res,
+        action='create_sibling_webdav',
+        status='notneeded',
+        type='dataset',
+        path=sub2.path
+    )
+    assert_in_results(
+        res,
+        action='add-sibling',
+        status='ok',
+        type='sibling',
+        path=sub2.path,
+        name='127.0.0.1',
+        url=dlaurl + 'sub2'
+    )
+    # subsub
+    assert_in_results(
+        res,
+        action='create_sibling_webdav',
+        status='notneeded',
+        type='dataset',
+        path=subsub.path
+    )
+    assert_in_results(
+        res,
+        action='create_sibling_webdav',
+        status='notneeded',
+        type='dataset',
+        path=subsub.path,
+    )
+
+    # should fail upfront with first discovered remote that already exist
+    assert_raises(IncompleteResultsError, ds.create_sibling_webdav, f'{url}',
+                  storage_sibling='yes', existing='error', recursive=True, **ca)
```

Comment:
I think it would be more useful to give `on_failure='ignore'` and test for the particular result (composition) that would trigger `IncompleteResultsError`. The former is relevant here, there latter is the present behavior of datalad core.

Reply:
Done